docs/reviewing.md

```diff
@@ -0,0 +1,15 @@
+# Reviewing policies
+
+The following policies are preferred during the review, but not mandatory:
+
+## Trailing white-spaces
+
+Avoid trailing white-space characters, if possible
+
+## Quotes
+
+If possible, try to avoid mixing single quotes (`'`) and double quotes (`"`) in python code (`conanfile.py`, `test_package/conanfile.py`). Consistency in preferred.
+
+## Order of methods and attributes
+
+Prefer alfa-numeric order of methods in python code (`conanfile.py`, `test_package/conanfile.py`), for instance, `configure` method should go before `package_info` method.
```

Comment:
I think there is a logical order for Conan methods, right? At least this is how I think about Conan recipes, it happens that for many of them the lexicographic ordering is equal to the logical one, but I prefer to use something meaningful.... otherwise, `requirements()` will be moved at the bottom of the `conanfile.py` (and I prefer to have it closer to the class attributes).

Reply:
@SpaceIm has proposed to use the order in which the methods are executed by Conan and I think that makes a lot of sense

Reply:
I would expect them to match the order they are documented, which probably should match the order they are executed (now doc has no order, just in case: https://docs.conan.io/en/latest/reference/conanfile/methods.html). Not sure if it is exactly the order they are executed (for example, and this is my very personal opinion, I prefer `package_id()` at the bottom, but it is usually executed when building the graph to search for available binaries).

Reply:
is there strict execution order? I think some methods could be called several times at different stages

Reply:
💭 I assume CCI only calls `create` to make the binaries.... could that be the order?

Reply:
not really. it first computes package ids to build and understand how many agents and of what type (win/linux/mac) to spawn. then it probably calls `create`. and afterwards, `upload` at some later stage.

Reply:
alpha-numeric order does not make sense.

Methods that work together should be grouped together.
Putting them in order that conan calls them makes total sense.

I also try to order the attributes at the top of a Conan file class:
- put name, description, license, homepage, topics at the top as these are user facing
- then url
- then options + default_options
- settings
- then all options that are build specific: options that must be set to make the package build.
Such as generators, no_copy_source, exports_sources, ...

The ordering of top attributes might be nitpicking though

Reply:
> not really. it first computes package ids to build and understand how many agents and of what type (win/linux/mac) to spawn. then it probably calls `create`. and afterwards, `upload` at some later stage.

I wouldn't take the conan-package-tools as a reference.
It does not fully initialize the recipe.
(early revisions of my boost pr at https://github.com/conan-io/conan-center-index/pull/2097 failed to build on conan-package-tools because of this)

Reply:
I don't mind doing an experiment (logging the order of methods executed on Conan Center Index CI) and documenting this order here.

Reply:
did some experiment in #3636. the current order of methods on CI:
- init
- set_name
- set_version
- export
- export_sources
- config_options
- configure
- requirements
- package_id
- build_id
- build_requirements
- system_requirements
- source
- imports
- build
- package
- package_info
never called on CI: 
- deploy
- test

should we document this order as preferred for recipes?
I don't mind, but keep in mind:
- conan implementation may change, and order of execution is subject to change as well (as long as it doesn't break, I think some methods could be safely swapped)
- CI implementation may change as well to call different commands in different order

Reply:
btw, another question, while we can have order of documented conanfile methods defined, what should we do with the following:
- private recipe methods (starting with `__`)
- so called `magic` methods (such as `__new__`, `__init__`, `__del__` and so on)
- protected recipe methods (starting with `_`)
- other (non-documented) public methods

in general, do we expect them to appear before/after documented `ConanFile` methods such as `source`?

Reply:
I don't think we should over-specify, having a default order for the Conan methods should cover most things already

Regarding custom methods as an example, in CMake based recipes we almost always have a `_cmake_configure()` method, where IMO it makes sense to have this directly above `build()`. Other private methods might not be that common and might only do little things like string manipulations/mappings. It might not always be that clear where to put them as in the case of `_cmake_configure()`. I would leave that to whoever is writing the recipe

Regarding enforcement, unfortunately I don't think we can add for this a nice config file like `.gitattributes` or `.editorconfig` which "just works" for most contributors, but if core contributors use this order + when we try to use this order in documentation and templates, then we will get it more standardized than before without annoying people too much for something non-functional.

Reply:
What @Croydon says. Let's recommend, not require or enforce.